runtime/ms-rest-azure/lib/cloudError.js

```diff
@@ -43,6 +45,10 @@ class CloudError extends Error {
         });
         this.details = tempDetails;
       }
+
+      if (parameters.innerError) {
+        this.innerError = JSON.parse(JSON.stringify(parameters.innerError));
```

Comment:
What is the need to stringify the object and then parse it again if it is already a JSON object. I understand that this is usually done to clone the object. However, I dont understand the need to do this in the current scenario.
Can you please explain?

Reply:
The details field is being reconstructed in every function. I assumed that we can assign other fields because the strings are immutable, and the complex objects should be deep cloned. I'm not too familiar with the code, so please let me know if the deep cloning is unnecessary.

Reply:
No worries. It is ok to keep it like that. Anyways serialize() and deserialize() functions in CloudError are for backwards compatibility. Newer versions of the generated sdk do not use them anymore. They only depend on the mapper and then serialize/deserialize accordingly.
So you are good for now.

Can you also update the Changelog.md (https://github.com/Azure/azure-sdk-for-node/blob/master/runtime/ms-rest-azure/Changelog.md)
 - Added `innererror` property to `CloudError`